### DISCUSSION TITLE
Porting our named pipe service to CoreWCF issue

### DISCUSSION THREAD

**Discussion by ahn25** [CoreWCF/CoreWCF#1488] [Q&A]
Hi,

We have ported our WCF named pipe service to .NET8 and CoreWCF and can connect from our .NET 4.8 client without any issues.

Example from the client side:

                using (var miscClient = ServiceClientProviderFactory.GenerateClientProvider().GetMiscClient())
                {
                     // Calling the service works fine
                     var itemList = miscClient.GetItemList();
                } <- exiting the using block will throw exceptions in CoreWCF.netFramingBase

When exiting the using block where the client connection is disposed/closed, the FramingModeHandshakeMiddleware.cs in the method OnConnectedCoreAsync method throws an exception due to the reuseHandler being null (class can be found in the project CoreWCF.NetFramingBase)

We can fix the problem, by doing the following:

![image](https://github.com/user-attachments/assets/75753b98-5530-487a-9537-cc1382dda84c)

But are we actually missing some configuration steps to not end in this scenario or is it a bug in CoreWCF?

Thanks for any input.

**Comment by mconnew** [comment]
Which versions are you using? I didn't release a 1.6.0 update for NetNamedPipe, and it's possible if you use the 1.6.0 version of NetFramingBase with the 1.5 version of NetNamedPipe that they might not work well together.

**Reply by mconnew** [reply]
I just took a look at the current code and the 1.6.0 diff, and that's exactly what happened. NamedPipeListenOptions derives from NetFraminListenOptions in 1.6.0, but not in 1.5. So listen options is null. I'll be releasing the next release soon and will release an updated preview version of NetNamedPipe which will solve this. In the mean time, use the latest 1.5.x version of everything.

**Reply by ahn25** [reply]
Thank you very much for the quick reply,  it is very much appreciated. We were using 1.6.0.

**Reply by MJE-GTI** [reply]
This also happens when you build the 1.6.0 branch locally because a code line is missing. I have created an issue for it.
https://github.com/CoreWCF/CoreWCF/issues/1492
